### DISCUSSION TITLE
Logic behind checks?

### DISCUSSION THREAD

**Discussion by alexander-winkler** [ChenglongMa/zoplicate#4] [General]
Hallo! I am thrilled about this Add-on as duplicate prevention in Zotero is something that I've been looking for for ages. The add-on doesn't work as anticipated, however. Would you mind describing the checks that are undertaken (which elements are compared, when are two objects considered duplicates) so that I can better understand how it works?

Thanks for your work!

**Comment by ChenglongMa** [comment]
Hi @alexander-winkler,

Thanks for your feedback!

I'm using the same logic as Zotero checks for duplicates.

As shown below, https://github.com/ChenglongMa/zoplicate/blob/c7ab500897edfce962f9eaa4a22eec5f48f82801/src/modules/notifier.ts#L12-L14

The logic behind this can be found in the official document:
https://www.zotero.org/support/duplicate_detection

> Zotero currently uses the the title, DOI, and ISBN fields to determine duplicates. If these fields match (or are absent), Zotero also compares the years of publication (if they are within a year of each other) and author/creator lists (if at least one author last name plus first initial matches) to determine duplicates. The algorithm will be improved in the future to incorporate other fields. 

Also, please let me know how you expect it to work. I can try to improve it.
Thanks!

**Reply by alexander-winkler** [reply]
Thanks for pointing me to the resource! I've tried a couple of duplicate imports and some checks failed for reasons I could not understand. I will investigate further and let you know.

**Reply by ChenglongMa** [reply]
Sure, no problem! 

_One tip_: 
Please check if the duplicate items can be found in the Zotero Duplicate panel.
If so, it should a bug of this add-on. Please let me know and I will fix it.
If not, it should be because the conditions for duplicates defined by Zotero are not met.

Hope this can help.
Thanks.
